source/mir/random/mixture.d

```diff
@@ -0,0 +1,112 @@
+/**
+Mixture methods that can be used to combine random variables.
+
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Ilya Yaroshenko, Sebastian Wilzbach
+*/
+
+module mir.random.mixture;
+
+import std.traits : isNumeric;
+
+/**
+_Discrete distribution sampler.
+Given an array of cumulative density points `cdPoints`,
+a _discrete distribution is defined.
+The cumulative density points can be calculated from probabilities or counts
+using `std.algorithm.cumulativeFold`. Hence `cdPoints` is assumed to be sorted.
+
+Params:
+    cdPoints = cumulative density points
+
+Returns:
+    A sampler that can be called to sample from the distribution.
+*/
+struct Discrete(T)
+    if (isNumeric!T)
+{
+    import std.range : SortedRange;
+
+    private const(T)[] _cdPoints;
+    private SortedRange!(const(T)[], "a <= b") r;
+
+    ///
+    this(const(T)[] cdPoints)
+    {
+        _cdPoints = cdPoints;
+        r = typeof(r)(_cdPoints);
+    }
+
+    /// cumulative density points
+    const(T)[] cdPoints() const @property
+    {
+        return _cdPoints;
+    }
+
+    /// samples a value from the discrete distribution
+    size_t opCall() const
+    {
+        import std.random : rndGen;
+        return opCall(rndGen);
+    }
+
+    /// samples a value from the discrete distribution using a custom random generator
+    size_t opCall(RNG)(ref RNG gen) const
+    {
+        import std.random : uniform;
+        T v = uniform(0, _cdPoints[$-1], gen);
```

Comment:
doesn't work - the prototype is `auto uniform(string boundaries = "[)", T1, T2)(T1 a, T2 b)`

https://dlang.org/phobos/std_random.html#.uniform